fastparse/src-3/fastparse/internal/MacroRepImpls.scala

```diff
@@ -0,0 +1,109 @@
+package fastparse.internal
+
+import fastparse.{Implicits, NoWhitespace, P, ParsingRun}
+
+import scala.annotation.tailrec
+import scala.quoted.*
+
+/** Implementations of the various `.rep`/`.repX` overloads. The most common
+  * and simple overloads are implemented as macros for performance, while the
+  * more complex/general cases are left as normal methods to avoid code bloat
+  * and allow the use of default/named arguments (which don't work in macros
+  * due to https://github.com/scala/bug/issues/5920).
+  *
+  * Even the normal method overloads are manually-specialized to some extent
+  * for various sorts of inputs as a best-effort attempt ot minimize branching
+  * in the hot paths.
+  */
+object MacroRepImpls {
+  def repXMacro0[T: Type, V: Type](
+      lhs: Expr[ParsingRun[T]],
+      whitespace: Null | Expr[ParsingRun[_] => ParsingRun[Unit]],
+      min: Null | Expr[Int],
+  )(repeater: Expr[Implicits.Repeater[T, V]], ctx: Expr[ParsingRun[_]])(using quotes: Quotes): Expr[ParsingRun[V]] = {
+    import quotes.reflect.*
+
+    '{
+
+      val ctx1        = $ctx
+      val repeater1   = $repeater
+      var originalCut = ctx1.cut
+      val acc         = repeater1.initial
+      @ _root_.scala.annotation.tailrec
```

Comment:
FWIW, you don't need to bother with fully-qualified names in Scala 3 macros, they're hygienic.

Reply:
LGTM!